cmd/calyptia/create_core_instance_aws.go

```
@@ -1,19 +1,203 @@
+	fs.StringSliceVar(&tags, "tags", nil, "Tags to apply to the core instance.")
+	fs.StringVar(&credentials, "credentials", "", "Path to the AWS credentials file. If not specified the default credential loader will be used.")
+	fs.StringVar(&profileFile, "profile-file", "", "Path to the AWS profile file. If not specified the default credential loader will be used.")
+	fs.StringVar(&profileName, "profile", "", "Name of the AWS profile to use, if not specified, the default profileFile will be used.")
+	// Set of parameters that map into https://docs.aws.amazon.com/sdk-for-go/api/service/ec2/#RunInstancesInput
+	fs.StringVar(&keyName, "key", awsClient.DefaultSecurityGroupName, "AWS Key to use for SSH into the core instance.")
+	fs.StringVar(&region, "region", awsClient.DefaultRegionName, "AWS region name to use in the instance.")
+	fs.StringVar(&instanceTypeName, "instance-type", awsClient.DefaultInstanceTypeName, "AWS Instance type to use (see https://aws.amazon.com/es/ec2/instance-types/) for details.")
+	fs.StringVar(&securityGroupName, "security-group", awsClient.DefaultSecurityGroupName, "AWS Security group name to use.")
+	fs.StringVar(&subnetID, "subnet-id", "", "AWS subnet name to use.If you don't specify a subnet ID, we choose a default subnet from your default VPC for you. If you don't have a default VPC, you MUST specify a subnet.")
+
+	// TODO: pass the environment name to the virtual machines.
+	_ = cmd.RegisterFlagCompletionFunc("environment", config.completeEnvironments)
+
 	return cmd
 }
+
+func ensureCoreInstanceName(ctx context.Context, config *config, name string) error {
+	_, err := getCoreInstanceByName(ctx, config, name)
+	if errors.Is(err, errCoreInstanceNotFound) {
+		return nil
+	}
+	return err
+}
+
+func getCoreInstanceByName(ctx context.Context, config *config, name string) (*types.Aggregator, error) {
+	coreInstances, _ := config.cloud.Aggregators(ctx, config.projectID, types.AggregatorsParams{
```

Why are you ignoring the error here?

---

Good one